### ISSUE TITLE
Document Package Superclasses specific to build systems

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#2606]
MakefilePackage, CMakePackage, AutotoolsPackage, etc.

**Comment by citibeth** [comment]
The new feature of `CMakePackage` that removes the need for `depends_on('cmake')` should also be documented here.

**Comment by citibeth** [comment]
@alalazo How do I convert to `AutotoolsPackage` for packages that do a test?  For example:
```
    def install(self, spec, prefix):
        configure('--prefix={0}'.format(prefix))

        make()
        if self.run_tests:
            make('check')
        make('install')
```

Thanks!

**Comment by adamjstewart** [comment]
Already answered this question in another PR, but you can just delete the entire `install` method. `AutotoolsPackage` basically looks the same by default.

**Comment by citibeth** [comment]
Addressed in #2780
